src/main/java/de/fraunhofer/aisec/crymlin/builtin/SplitSubset.java

```diff
@@ -0,0 +1,81 @@
+
+package de.fraunhofer.aisec.crymlin.builtin;
+
+import de.fraunhofer.aisec.analysis.markevaluation.ExpressionEvaluator;
+import de.fraunhofer.aisec.analysis.markevaluation.ExpressionHelper;
+import de.fraunhofer.aisec.analysis.structures.*;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+
+/**
+ * Method signature: _split_subset(String str, String splitter, List superset)
```

Comment:
I would suggest something like ´_split_match_subset` to make it clear, that this is a matching function. Maybe even `_match_split_subet`

Reply:
or can we even merge this with the other one and have a fourth parameter that configures the "matching"? ALL vs. SUBSET or something?